.github/workflows/Intel.yml

```diff
@@ -27,60 +21,29 @@ jobs:
 
     steps:
 
-    - uses: actions/checkout@v2
+    - name: checkout
+      uses: actions/checkout@v4
+      with:
+        submodules: true
 
       # See https://software.intel.com/content/www/us/en/develop/articles/oneapi-repo-instructions.html
-    - name: install-intel
-      run: |
-        cd /tmp
-        wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
-        sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
-        rm GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
-        echo "deb https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
-        sudo apt-get update
-        sudo apt-get install intel-oneapi-dev-utilities intel-oneapi-openmp intel-oneapi-mpi intel-oneapi-mpi-devel intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic intel-oneapi-compiler-fortran
-        echo "source /opt/intel/oneapi/setvars.sh" >> ~/.bash_profile
-        if [ ${{ matrix.compilers }} == "oneapi" ]; then
-          echo "export CC=mpiicx FC=mpiifx" >> ~/.bash_profile
-        elif [ ${{ matrix.compilers }} == "intel" ]; then
-          echo "export CC=mpiicc FC=mpiifort" >> ~/.bash_profile
-        fi
-
-    - name: checkout-bacio
-      uses: actions/checkout@v2
+    - name: "Install Intel"
+      uses: NOAA-EMC/ci-install-intel-toolkit@develop
       with:
-        repository: NOAA-EMC/NCEPLIBS-bacio
-        path: bacio
-        ref: develop
+        install-mpi: true
+        mpi-wrapper-setup: ${{ matrix.compilers }}
 
-    - name: build-bacio
-      run: |
-        cd bacio
-        mkdir build
-        cd build
-        cmake .. -DCMAKE_INSTALL_PREFIX=~/bacio
-        make -j2
-        make install
-
-    - name: checkout-w3emc
-      uses: actions/checkout@v2
+    - name: "Build dependencies"
+      uses: NOAA-EMC/ci-build-nceplibs@develop
       with:
-        repository: NOAA-EMC/NCEPLIBS-w3emc
-        path: w3emc
-        ref: develop
-
-    - name: build-w3emc
-      run: |
-        cd w3emc
-        mkdir build
-        cd build
-        cmake .. -DCMAKE_INSTALL_PREFIX=~/w3emc -DCMAKE_PREFIX_PATH="~/" -DBUILD_WITH_BUFR=OFF
-        make -j2
-        make install
+        bacio-version: develop
+        w3emc-version: develop
+        w3emc-cmake-args: -DBUILD_WITH_BUFR=OFF
+        key-prefix: Intel-
+        key-suffix: -1
```

Comment:
Not sure I understand this.

Reply:
The Actions for building packages (ci-build-nceplibs and ci-build-cmake-code) have caching capability built in, and have a built-in naming scheme. The `key-prefix` and `key-suffix` options are ways to modify the cache keys to distinguish between workflows, between different 'matrix' options within a workflow, etc. So in this case, I'm prepending 'Intel-' to distinguish the cache entries for the Intel-built packages (see https://github.com/NOAA-EMC/NCEPLIBS-nemsio/actions/caches). The '-1' suffix here isn't really necessary at the moment, I just put it there since usually at some point somebody wants to start over with a clean cache by adding -1, -2, ... at the end of the cache key.